src/Build/Logging/ParallelLogger/ParallelConsoleLogger.cs

```diff
@@ -535,6 +532,58 @@ public override void ProjectStartedHandler(object sender, ProjectStartedEventArg
                     WriteItems(e, e.Items);
                 }
             }
+
+            //node and project context ids for the propertyOutputMap key
+            int node = -1;
+            int projectContextId = -1;
+            if (e.BuildEventContext != null)
+            {
+                node = e.BuildEventContext.NodeId;
+                projectContextId = e.BuildEventContext.ProjectContextId;
+            }
+            //creating the value to be added to the propertyOutputMap
+            StringBuilder LogOutputProperties = new StringBuilder();
+            if (e.BuildEventContext != null && e.Items != null)
+            {
+                foreach (DictionaryEntry item in e.Items)
```

Comment:
Do you have to foreach here instead of doing a dictionary lookup like `e.Items["LogOutputProperties"]`?

Reply:
Ah, I see you do because it's a non-generic `IEnumerable` 😱